charts/argo-cd/templates/argocd-application-controller/servicemonitor.yaml

```diff
@@ -3,29 +3,34 @@ apiVersion: monitoring.coreos.com/v1
 kind: ServiceMonitor
 metadata:
   name: {{ template "argo-cd.controller.fullname" . }}
-  {{- if .Values.controller.metrics.serviceMonitor.namespace }}
-  namespace: {{ .Values.controller.metrics.serviceMonitor.namespace }}
+  {{- with .Values.controller.metrics.serviceMonitor.namespace }}
+  namespace: {{ . }}
   {{- end }}
   labels:
     {{- include "argo-cd.labels" (dict "context" . "component" .Values.controller.name "name" .Values.controller.name) | nindent 4 }}
-    {{- if .Values.controller.metrics.serviceMonitor.selector }}
-{{- toYaml .Values.controller.metrics.serviceMonitor.selector | nindent 4 }}
+    {{- with .Values.controller.metrics.serviceMonitor.selector }}
+      {{- toYaml . | nindent 4 }}
     {{- end }}
-    {{- if .Values.controller.metrics.serviceMonitor.additionalLabels }}
-{{- toYaml .Values.controller.metrics.serviceMonitor.additionalLabels | nindent 4 }}
+    {{- with .Values.controller.metrics.serviceMonitor.additionalLabels }}
+      {{- toYaml . | nindent 4 }}
     {{- end }}
 spec:
   endpoints:
     - port: metrics
-      {{- with .Values.controller.metrics.serviceMonitor.interval }}
-      interval: {{ . }}
-      {{- end }}
       path: /metrics
+      interval: {{ .Values.controller.metrics.serviceMonitor.interval }}
```

Comment:
I think we should leave the `with` conditional around interval. Some users may want to unset this and use the global Prometheus scrape interval. WDYT?

Reply:
You are right. My mistake - I thought this was there for backwards compatibility when this feature was introduced and 30s is a default for ServiceMonitor CRD. Will revert so global scrape interval can be used.

Reply:
Another option would be to use 1min interval which is a [default](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config) for Prometheus.

Reply:
I think set it to 1min is not a real solution. The default interval is configurable in every prometheus instance :-)
Eg. the kube-prometheus-stack has 30s default
```console
/prometheus $ cat /etc/prometheus/config_out/prometheus.env.yaml
global:
  evaluation_interval: 30s
  scrape_interval: 30s
# (..)
```

Reply:
Thanks for extending my knowledge that this differs even across stacks :)